src/main/java/teammates/logic/api/Logic.java

```diff
@@ -785,16 +786,20 @@ public static boolean isCourseArchived(CourseAttributes course, InstructorAttrib
     /**
      * Preconditions: <br>
      * * All parameters are non-null.
+     * <br>
+     * The second parameter ctaMap is used to keep the query results in the form of
+     * course ID -> {@link StudentAttributes} mapping so that they can be re-used.
      * 
      * @return Details of courses the student is in. CourseData objects
      *         returned contain details of feedback sessions too (except the ones
      *         still AWAITING).
      */
-    public List<CourseDetailsBundle> getCourseDetailsListForStudent(String googleId)
+    public List<CourseDetailsBundle> getCourseDetailsListForStudent(String googleId,
+                                                                    Map<String, StudentAttributes> ctaMap)
```

Comment:
After more examination, it looks like you are using an input parameter to retrieve output? That's 'sneaky'. I'd rather not do 'sneaky' stuff in the code. Having a return value adds to the confusion. May be define a new return data type that contains all the values you need?